fix: mobile issue with form pop up

safwansamsudeen opened the pull request:
<!--

Some key notes before you open a PR:

 1. Select which branch should this PR be merged in?
 2. PR name follows [convention](http://karma-runner.github.io/4.0/dev/git-commit-msg.html)
 3. 
 4. All tests pass locally, UI and Unit tests
 5. All business logic and validations must be on the server-side
 6. Update necessary Documentation
 7. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes


Also, if you're new here

- Documentation Guidelines => https://github.com/frappe/erpnext/wiki/Updating-Documentation

- Contribution Guide => https://github.com/frappe/frappe/blob/develop/.github/CONTRIBUTING.md

- Pull Request Checklist => https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

-->

> Please provide enough information so that others can review your pull request:

Closes #24049.

> Explain the **details** for making this change. What existing problem does the pull request solve?

On mobile view, certain pop ups resulted in scrunching of the layout. With this update, the content is rendered correctly.

> Screenshots/GIFs

Desktop:
<img width="564" alt="image" src="https://github.com/frappe/frappe/assets/62411302/b174c03c-7b0a-409a-bb9a-b75761dbbae4">

Mobile (previous):
<img width="226" alt="image" src="https://github.com/frappe/frappe/assets/62411302/5084fdfd-623b-437a-840c-048f83fb2df5">


Mobile (now):
<img width="311" alt="image" src="https://github.com/frappe/frappe/assets/62411302/272be0a7-2455-4ff3-b016-aadc5c5d4569">

<!-- Add images/recordings to better visualize the change: expected/current behviour -->

safwansamsudeen committed:
fix: mobile configure columns issue

mohsinalimat commented:
> Mobile (now): <img alt="image" width="311" src="https://private-user-images.githubusercontent.com/62411302/294175980-272be0a7-2455-4ff3-b016-aadc5c5d4569.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzNzEyMjMsIm5iZiI6MTcwNDM3MDkyMywicGF0aCI6Ii82MjQxMTMwMi8yOTQxNzU5ODAtMjcyYmUwYTctMjQ1NS00ZmYzLWIwMTYtYWFkYzVjNWQ0NTY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDEyMjIwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ5NjNmYmIyZGE2ZTAyMTE3MDQ3NzhhNjczN2E2ODg5YzRkYjUwZTk0MzIwYzAwNGI2YWEzODZiYzM0NjI0OGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WnTTGyOo7V3oh1T0un4PpMuCNnhsgAuJESDqZRJmOwo">

Text and button not in center.

ankush commented:
@safwansamsudeen congrats on first PR! That was quite fast :tada: 

Here's how it looks on desktop and that's roughly how it should look on mobile too. 

Make the middle column flexible so whitespace is reduced in mobile view. 

![image](https://github.com/frappe/frappe/assets/9079960/3cd22e37-946e-4306-ad31-e13e18b85f0e)

safwansamsudeen committed:
fix: correct grid view issue

safwansamsudeen commented:
It turns out my original commit didn't fix the exact issue I had referenced - the Configure Columns pop up uses the Grid View, while I had fixed the List view (other pop ups use it, including the one I'd shown).

I've fixed the Grid view now - but a quick check through [the possible views and their files](https://frappeframework.com/docs/user/en/desk) shows that even Kanban view should have the same problem. I'm not sure where users will encounter it - should I fix it anyway?

(Programming nit, not too important: for some reason, in the Grid view correction, the maximum number of Bootstrap columns in mobile has to be 11. If I correctly fill it up with 12, the layout breaks. Any ideas why? I'm thinking it might have something to do with the FA delete icon).

Current view:
<img width="303" alt="image" src="https://github.com/frappe/frappe/assets/62411302/12aab102-191c-4055-88b6-63eb413ef7d7">

safwansamsudeen committed:
fix header too
